core-bundle/src/EventListener/DataContainer/PageRoutingListener.php

```diff
@@ -54,14 +54,13 @@ public function generateRouteConflicts(DataContainer $dc): string
             return '';
         }
 
-        $aliasPages = $pageAdapter->findSimilarByAlias($currentPage);
-
-        if (null === $aliasPages) {
+        if (!$aliasPages = $pageAdapter->findSimilarByAlias($currentPage)) {
             return '';
         }
 
         $conflicts = [];
-        $currentUrl = $this->buildUrl($currentPage->alias, $currentPage->urlPrefix, $currentPage->urlSuffix);
+        $currentRoute = $this->pageRegistry->getRoute($currentPage);
+        $currentUrl = $this->buildUrl($currentPage->alias, $currentRoute->getUrlPrefix(), $currentRoute->getUrlSuffix());
```

Comment:
This would make the `buildUrl()` method obsolete and would take the actual path from `AsPage()` attribute into consideration. Would need to be applied for `$aliasUrl` as well obviously.

```suggestion
        $currentUrl = $currentRoute->compile()->getStaticPrefix().$currentRoute->getUrlSuffix();
```

Reply:
see https://github.com/leofeyer/contao/pull/16